htmresearch/algorithms/temporal_memory_phases.py

```diff
@@ -350,7 +350,9 @@ def learnOnSegments(self,
                           self.permanenceDecrement)
 
       if isLearningSegment:
-        n = self.maxNewSynapseCount - len(activeSynapses)
+        n = min(self.maxNewSynapseCount,
```

Comment:
@mrcslws Note that I did make the adding synapses bug fix in this implementation, as it is needed to form unions.